Python/specialize.c

```diff
@@ -637,3 +640,103 @@ _Py_Specialize_LoadGlobal(
     cache0->counter = saturating_start();
     return 0;
 }
+
+/* TODO:
+    - Specialize calling C types like int() with CALL_CTYPE
+    - Specialize python function calls.
+*/
+int
+_Py_Specialize_CallFunction(PyObject *builtins,
+    PyObject **stack_pointer, uint8_t original_oparg,
+    _Py_CODEUNIT *instr, SpecializedCacheEntry *cache)
+{
+    PyObject *callable = stack_pointer[-(original_oparg + 1)];
+    _PyAdaptiveEntry *cache0 = &cache->adaptive;
+    PyTypeObject *type = Py_TYPE(callable);
+    /* Specialize C functions */
+    if (PyCFunction_CheckExact(callable)) {
+        PyCFunctionObject *meth = (PyCFunctionObject *)callable;
+        if (PyCFunction_GET_FUNCTION(callable) == NULL) {
+            goto fail;
+        }
+        /* Currently not optimizing anything that is FASTCALL, has keywords, has
```

Comment:
This comment is likely to end up incorrect as we change the code, and doesn't say anything that isn't clear from the code.
Probably best to remove it.